service/src/main/java/uk/nhs/adaptors/gp2gp/common/storage/StorageConnectorService.java

```diff
@@ -0,0 +1,36 @@
+package uk.nhs.adaptors.gp2gp.common.storage;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import static uk.nhs.adaptors.gp2gp.gpc.GpcFileNameConstants.GPC_STRUCTURED_FILE_EXTENSION;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import lombok.AllArgsConstructor;
+import uk.nhs.adaptors.gp2gp.gpc.GpcStructuredResponseObject;
+
+@Service
+@AllArgsConstructor
+public class StorageConnectorService {
+    private final StorageConnectorFactory storageConnectorFactory;
+    private final ObjectMapper objectMapper;
+
+    @SuppressFBWarnings(
+        value = "NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE",
+        justification = "SpotBugs issue with fix not yet released https://github.com/spotbugs/spotbugs/issues/456")
+    public void handleStructuredRecord(GpcStructuredResponseObject response) throws IOException {
+        String jsonStringResponse = objectMapper.writeValueAsString(response);
```

Comment:
Found your IOException. It would be quite the exception indeed if this writeValueAsString failed with a JSONProcessingException. You would not expect the serialised of a simple internal object to fail unless there is a real programming error.

I would annotate `handleStructuredRecord` with `@SneakyThrows(JsonProcessingException.class)`. I think in this case suppressing the checked exception can be justified.